ci_scripts/install-tools.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+# Purpose: Installs or upgrades essential development tools.
+# Notes:
+#   - Supports macOS and Linux (best effort) for installations via `brew install` and `go install`.
+#   - Installs tools like kubectl, helm, pre-commit, go, and chloggen.
+#   - Use OVERRIDE_OS_CHECK=true to bypass OS compatibility checks.
+#   - This script is intended to be run via `make install-tools`.
+#   - Prompts the user for approval to install or update each tool if the tool is out of date.
+#
+# Example Usage:
+#    make install-tools [OVERRIDE_OS_CHECK=true]
+
+# Function to install or upgrade a tool
+install_or_upgrade() {
+  local tool=$1
+  local type=$2
+
+  case $type in
+    brew)
+      install_or_upgrade_brew "$tool"
+      ;;
+    go)
+      install_or_upgrade_go "$tool"
+      ;;
+    *)
+      echo "Unsupported tool type: $type"
+      exit 1
+      ;;
+  esac
+}
+
+# Function to install or upgrade a Homebrew-based tool
+install_or_upgrade_brew() {
+  local tool=$1
+  local installed_version=$(brew list $tool --versions | awk '{print $2}')
+  local latest_version=$(brew info --json=v1 "$tool" | jq -r '.[0].versions.stable')
```

Comment:
Will we be upgrading to the latest version available for helm at all times?

Reply:
While most tool dependencies in this project aren't pinned to specific versions, we've found that using the latest versions generally works without issues. However, older dependencies might not support newer features. Although this might only present challenges in rare instances, it's something we can address as needed in the future.

Reply:
At this time I would recommend installing latest. If we invest more into developer tool version control I think we can pin the version for the involved tools. Done.